src/mpid/ch4/generic/mpidig_globals.c

```diff
@@ -53,7 +60,10 @@ void MPIDIG_comm_abort(MPIR_Comm * comm, int exit_code)
 
   fn_exit:
     MPIR_FUNC_VERBOSE_EXIT(MPID_STATE_MPIDIG_COMM_ABORT);
-    MPL_exit(exit_code);
+    if (comm->comm_kind == MPIR_COMM_KIND__INTRACOMM)
+        MPL_exit(exit_code);
+    else
+        return MPI_SUCCESS;
```

Comment:
I mentioned this yesterday, but now I get why you made this function return `int`. We probably should change it back from `void` if there is the option of not exiting the calling process. Sorry for the churn.